variables.tf

```diff
@@ -0,0 +1,82 @@
+variable "resource_group" {
+  description = "Azure resource group."
+  type        = string
+  default     = "aap_on_azure"
+}
+
+variable "location" {
+  description = "Azure location."
+  type        = string
+  default     = "East US"
+}
+
+variable "deployment_id" {
+  description = "Creates a random string that will be used in tagging for correlating the resources used with a deployment of AAP."
+  type    = string
+  validation {
+    condition     = (length(var.deployment_id) == 8 || length(var.deployment_id) == 0) && (can(regex("^[a-z]", var.deployment_id)) || var.deployment_id == "")
+    error_message = "deployment_id length should be 8 chars and should contain lower case alpha chars only"
+  }
+}
+
+variable "infrastructure_vpc_cidr" {
+  description = <<-EOT
+    IPv4 CIDR netmask for the VPC resource.
+  EOT
+  type        = string
+  default     = "172.16.0.0/22"
+}
+
+variable "infrastructure_vpc_subnet_cidr_controller-eda-hub" {
+  description = <<-EOT
+    IPv4 CIDR netmask for the subnet resource for controller, hub and EDA nodes.
+  EOT
+  type        = string
+  default     = "172.16.0.0/24"
+}
+
+variable "infrastructure_vpc_subnet_cidr_execution" {
+  description = <<-EOT
+    IPv4 CIDR netmask for the subnet resource for execution nodes.
+  EOT
+  type        = string
+  default     = "172.16.1.0/24"
+}
+
+variable "infrastructure_vpc_subnet_cidr_postgres" {
+  description = <<-EOT
+    IPv4 CIDR netmask for the subnet resource for postgres.
+  EOT
+  type        = string
+  default     = "172.16.2.0/24"
+}
+
+variable "infrastructure_db_username" {
+  description = "PostgreSQL username."
+  type        = string
+  default     = "psqladmin"
```

Comment:
I would suggest not defaulting the password and make it a required input from the user, like the Ansible repo does.